src/Validation/MailValidator.php

```diff
@@ -1,14 +1,18 @@
 <?php
 
-namespace WMDE\Fundraising\Frontend;
+namespace WMDE\Fundraising\Frontend\Validation;
+
+use WMDE\Fundraising\Frontend\MailAddress;
 
 /**
  * @licence GNU GPL v2+
  * @author Christoph Fischer < christoph.fischer@wikimedia.de >
+ * @author Gabriel Birke < gabriel.birke@wikimedia.de >
  */
-class MailValidator {
+class MailValidator implements ScalarValueValidator {
```

Comment:
The way this validator works it is actually also a required field validator. There is no way to make the email address an optional field, because passing an invalid value like an empty string, or null, always makes the validation fail.